src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -356,6 +357,14 @@ private static Uri CreateConnectUrl(Uri url, string connectionId)
             Log.TransportStarted(_logger, _transport);
         }
 
+        private void completePipePair(DuplexPipe.DuplexPipePair duplexPipePair)
```

Comment:
Pass the two pipes in separately. I don't like having `DuplexPipe.DuplexPipePair` here :). Just a style thing.

Reply:
Pulled the complete of the pipes out of the method entirely.